.github/workflows/predict.yml

```diff
@@ -18,109 +18,62 @@ on:
+        uses: actions/checkout@v4
 
-      - name: Split library by partition variables
+      - name: Delete unnecessary folders
         run: |
-          numerator=${{ inputs.numerator }}
-          denominator=${{ inputs.denominator }}
-          if [[ ${{ inputs.sample-only }} ]]; then
-            filename="reference_library_${{ inputs.sample-only }}.csv"
-          else
-            filename="reference_library.csv"
-          fi
-          
-          ## Total # lines in given reference library
-          total_lines=$(wc -l < "$filename")
-          echo $total_lines
-          
-          # Determine the # of file lines per partition
-          ((lines_per_file = (total_lines + $denominator-1) / $denominator))
-
-          # Split the file
-          split -l $lines_per_file -a 4 -d --additional-suffix=.csv "$filename" partition_
-          
-          # Add 'smiles' to header of all partition files beyond the first
-          for (( i=1; i<$denominator; i++ ))
-          do
-            partition=$(printf "partition_%04d.csv" $i)
-            sed -i '1i smiles' "$partition"
-          done
-
-          # List files in dir
-          ls
+          shopt -s extglob
+          rm -rf !(scripts)
 
-      - name: Add conda to system path
-        run: echo $CONDA/bin >> $GITHUB_PATH
-
-      - name: Source conda
-        run: source $CONDA/etc/profile.d/conda.sh
-
-      - name: Set Python to 3.10.10
-        run:
-         conda install -y python=3.10.10
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@master
+        with:
+            aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY }}
+            aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+            aws-region: eu-central-1
 
+      - name: Set up Python environment
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.10'
+      
       - name: Install dependencies
         run: |
-          source activate
-          conda init
-          conda install git-lfs -c conda-forge
-          git-lfs install
-          conda install gh -c conda-forge
-          
-      - name: Install ersilia
-        run: |
-          source activate
-          python --version
-          echo "After conda init"
-          conda init
-          python -m pip install -e .[test]
-          
-      - name: Predict partition output and upload
-        run: |
-          numerator=${{ inputs.numerator }}
-          index=$((numerator - 1))
-          source activate
-          echo "Sample model id selected: ${{ inputs.model-id }}"
-          ersilia fetch ${{ inputs.model-id }}
-          ersilia serve ${{ inputs.model-id }}
-          echo "${{ inputs.model-id }} succesfully fetched and served"
-          ersilia api -i $(printf "partition_%04d.csv" $index) -o ../${{ inputs.SHA }}$(printf "_%04d.csv" $numerator)
+          python -m pip install --upgrade pip
+          pip install boto3 pandas ersilia
```

Comment:
use `make install` instead

Reply:
https://github.com/ersilia-os/model-inference-pipeline/pull/29/commits/e54b85b0574a66e77df0a6b13874e3828c763243 let me know if any concerns with this

Reply:
Made a commit for this, but perhaps let's resolve the pandas/numpy issues here (https://github.com/ersilia-os/model-inference-pipeline/actions/runs/10233796760) first?